### PR TITLE
fix: remove unverified acP/genP/gridP power adjustment, bump min HA version

### PR DESCRIPTION
**Pull request by Veldkornet**
# 🛰️ HYXi Cloud  PR

## 📝 Description
Removes the unverified acP/genP/gridP power adjustment (a subtraction against a raw `acl` metric plus fixed 0.96/2.0 multipliers, with no cited source), collapses the acE/gridF/batTmp same-quantity fallbacks into a declarative table, adds missing test coverage for two of those fallbacks, and bumps the minimum supported Home Assistant version to 2026.3.0.

## 💡 Rationale
While investigating why the new microinverter aggregate sensors (#598) report "unknown," a live hybrid-inverter debug log (poll + push payloads, 168 raw keys) showed the `acl` field the acP/genP/gridP adjustment depends on never actually appears — confirmed separately against hyxi-cloud-api's bundled example data too. No `SensorEntityDescription` for `genP` even exists, so that branch could never instantiate a sensor in the first place. The multipliers had no cited source (vendor doc, issue, or commit rationale) and landed in an unrelated lint/tooling PR (#309); they were silently under-reporting acP by ~4% with nothing behind the correction. Removed — acP/gridP now report the raw metric unmodified.

The remaining acE→efpv, gridF→f, and batTmp→batTch fallbacks are real, PR-cited substitutions (one raw field standing in for another that carries the *same* physical quantity, no scaling factor) — kept, and now documented inline with which PR introduced each and why. They were three copy-pasted `if key == X and device_type in (...)` blocks; now that there's a third one, collapsed them into one `_SAME_QUANTITY_FALLBACKS` table so a future fallback is a data entry instead of another copied block. Added dedicated test coverage for the gridF/batTmp entries, which previously had none (only acE had a behavior test) — including confirming they do *not* treat `0.0` as missing, unlike acE.

Also bumped `hacs.json`'s minimum Home Assistant version to 2026.3.0 (the first Core release running on Python 3.14). The codebase already requires PEP 758 (unparenthesized multi-exception `except` clauses — enforced repo-wide by this project's own ruff-format config plus a dedicated pre-commit hook) and CI already tests only against 3.14, but `hacs.json` still advertised support back to 2024.1.0, which would hit a hard `SyntaxError` importing `sensor.py` on any older Python.

**Heads up for release notes:** anyone with the `acP` sensor in HA's Energy dashboard/long-term statistics will see a ~4.2% step increase the moment this ships. That's the corrected value, not a regression, but nothing in the entity metadata explains the jump on its own.

<details>
<summary><b>🛠️ Change Manifest (Technical)</b></summary>

- [x] Automated tests (Pytest/Hypothesis) must pass for merge.
- [x] Linting requirements (Ruff) must be met.
- [x] Logic impacts `custom_components`?
- [ ] Logic impacts `api_client`?
</details>

---

## 📸 Validation & Logs
- `pytest tests/ --ignore=tests/integration` → 463 passed, 1 skipped
- `HYXI_INTEGRATION_TEST=1 pytest tests/integration` → 32 passed
- All pre-commit hooks (ruff, ruff-format, mypy, pylint, vulture, codespell, secret scan, exception-parenthesization check) passed on both commits.

## 🔗 Traceability
- **Relates to:** #598
- **Closes:**

## Summary by Sourcery

Remove unverified power corrections, streamline valid metric fallbacks, and improve diagnostics for unavailable microinverter aggregates.

Bug Fixes:
- Restore raw acP and gridP sensor values by removing unsupported metric adjustments and scaling factors.
- Improve diagnostics for microinverter aggregate sensors by logging unavailable values once at debug level and resetting the log latch when data returns.

Enhancements:
- Consolidate same-quantity metric fallbacks into a declarative mapping while preserving device-specific behavior and zero-value handling.
- Add robust numeric-zero detection for fallback processing.

Tests:
- Expand sensor tests for gridF and batTmp fallbacks, raw acP passthrough, zero-value behavior, and microinverter diagnostic logging.

Chores:
- Raise the minimum advertised Home Assistant version to 2026.3.0.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved handling of missing, zero, and equivalent metric values for more reliable sensor readings.
  - Corrected power readings by removing unintended adjustments and scaling.
  - Improved microinverter aggregate sensor diagnostics, including clearer notifications when no usable data is available.
  - Diagnostic alerts now avoid duplicates and automatically reset when valid data returns.

### CODE REVIEW

File: tests/test_sensor_logic.py [hunk 1]
```diff
@@ -1026,45 +1026,6 @@ def mock_async_add_entities(entities):
...
-    assert sensor._get_metric_float("invalid") is None
-    assert sensor._get_metric_float("none") is None
-    assert sensor._get_metric_float("missing") is None
-    assert sensor._get_metric_float("type_error") is None
-    assert sensor._get_metric_float("list_error") is None
-
-
 @pytest.mark.asyncio
 async def test_new_telemetry_keys_registration_and_parsing():
     """Verify that all 29 new telemetry/Micro ESS sensors are registered and cast correctly."""
```
[the remark this answers is not in the document]
Comment (Veldkornet) [reply tests/test_sensor_logic.py]: Fixed — added tests in tests/test_micro_inverter.py covering the debug log firing once while stuck with no usable value, not firing when DEBUG is disabled, and the dedup latch resetting once a usable value reappears (so logging resumes if it goes missing again). Your suggested implementation referenced a different project's module path (custom_components.hoymiles.sensor / AggregatedMicroinverterSensor), so I wrote the tests fresh against the actual HyxiMicroinverterSumSensor using caplog.

File: custom_components/hyxi_cloud/const.py [hunk 2]
```diff
@@ -91,6 +91,19 @@ def is_null_value(value: Any) -> bool:
...
+    """Check if a value is numerically zero, regardless of its exact string/type form.
+
+    Unlike a literal `str(value) == "0.0"` check, this matches ints, "0",
+    "0.00", "-0.0", etc. Non-numeric/unparseable values are treated as
+    not-zero rather than raising.
+    """
+    try:
+        return not float(value)
+    except TypeError, ValueError:
+        return False
```
[the remark this answers is not in the document]
Comment (Veldkornet) [reply custom_components/hyxi_cloud/const.py]: Thanks — the OverflowError gap was real, fixed (added to the except tuple, plus a test covering it with a huge int).

The "invalid Python 3 syntax" framing isn't accurate for this repo though: unparenthesized `except X, Y:` is valid as of PEP 758 (Python 3.14), and this project already enforces exactly that style repo-wide — its own ruff-format config plus a dedicated pre-commit hook literally named "Check for parenthesized exceptions (use: except Exc1, Exc2:)" reformat it back to unparenthesized on every commit. Your script probably ran on an older Python than this project targets — this same PR bumps hacs.json's minimum HA version to 2026.3.0 specifically because it's the first Core release on Python 3.14, which the codebase (and CI) already require.

File: tests/test_sensor_logic.py [hunk 3]
```diff
@@ -1395,50 +1362,74 @@ def test_hyxi_sensor_advanced_mappings(base_sensor):
...
             },
         }
     }
 
     # 1. Parent Sn via_device link
-    sensor.entity_description.key = "genP"
+    sensor.entity_description.key = "acP"
     sensor._sn = "INV123"
     sensor._dev_data = coordinator.data["INV123"]
     sensor._metrics = sensor._dev_data["metrics"]
```
[the remark this answers is not in the document]
Comment (Veldkornet) [reply tests/test_sensor_logic.py]: Good catch — fixed. Added the missing update+assert for acP via _handle_coordinator_update(), so the removed adjustment (acP passing through unmodified now) is actually covered by a regression test instead of just being claimed in the docstring.
